plugins/workflow/plugin.go

```diff
@@ -24,40 +23,69 @@ const (
 
 // Plugin manages workflows and workers.
 type Plugin struct {
-	temporal  temporal.Temporal
-	events    util.EventsHandler
-	server    server.Server
-	log       log.Logger
-	mu        sync.Mutex
-	lastReset time.Time
-	pool      workflowPool
+	temporal temporal.Temporal
+	events   util.EventsHandler
+	server   server.Server
+	log      log.Logger
+	mu       sync.Mutex
+	reset    chan struct{}
+	pool     workflowPool
+	closing  int64
 }
 
 // logger dep also
 func (svc *Plugin) Init(temporal temporal.Temporal, server server.Server, log log.Logger) error {
 	svc.temporal = temporal
 	svc.server = server
+	svc.events = util.NewEventsHandler()
 	svc.log = log
+	svc.reset = make(chan struct{})
+
 	return nil
 }
 
 // Serve starts workflow service.
 func (svc *Plugin) Serve() chan error {
 	errCh := make(chan error, 1)
 
-	pool, err := svc.initPool()
+	pool, err := svc.startPool()
 	if err != nil {
-		errCh <- errors.E("initPool", err)
+		errCh <- errors.E("startPool", err)
 		return errCh
 	}
 
 	svc.pool = pool
 
+	go func() {
+		for {
+			select {
+			case <-svc.reset:
+				if atomic.LoadInt64(&svc.closing) == 1 {
```

Comment:
Just to think about. Case: `atomic.LoadInt64(&svc.closing) is equal to 0` then by some signal (endure send Stop) value of `svc.closing` became 1, but in this loop you assuming, that `svc.closing` is still 1. This place look like dangerous for me :)